tools/check_isa.cpp

```diff
@@ -10,246 +10,63 @@
-        bool clx = skx && avx512_vnni;
-        [[maybe_unused]] bool cpx = clx && avx512_bf16;
-        bool icl =
-            clx && avx512_vbmi2 && avx512_gfni && avx512_vaes && avx512_vpclmulqdq && avx512_bitalg && avx512_vpopcntdq;
-        [[maybe_unused]] bool tgl = icl && avx512_vp2intersect;
-        bool spr =
-            icl && avx512_bf16 && avx512_amx_bf16 && avx512_amx_tile && avx512_amx_int8 && avx_vnni && avx512_fp16;
-        // According to spec Gramnite Rapids is AVX 10.1 but in LLVM it's SPR with two additional features.
-        [[maybe_unused]] bool gnr = spr && amxfp16 && prefetchi;
-
-        bool avx10 = (info3[3] & (1 << 19)) != 0;
-
-        if (avx10) {
-            // clang-format off
-
-            int info_avx10[4] = {0, 0, 0, 0};
-            if (max_level >= 24) {
-                __cpuidex(info_avx10, 0x24, 0);
-            }
-            int avx10_ver = info_avx10[1] & 0xFF;
-            bool avx10_256 = (info_avx10[1] & (1 << 17)) != 0;
-            bool avx10_512 = (info_avx10[1] & (1 << 18)) != 0;
-            [[maybe_unused]] bool avx10_1_256 = avx10_256 && (avx10_ver >= 1);
-            [[maybe_unused]] bool avx10_1_512 = avx10_512 && (avx10_ver >= 1);
-            [[maybe_unused]] bool avx10_2_256 = avx10_256 && (avx10_ver >= 2);
-            [[maybe_unused]] bool avx10_2_512 = avx10_512 && (avx10_ver >= 2);
-            // clang-format on
-
-            // Diamond Rapids:         DMR = GNR + AVX10_2_512 + APX + ... (For the whole list see x86TargetParser.cpp)
-            bool dmr = gnr && avx10_2_512 && apx && cmpccxadd && avxneconvert && avxifma && avxvnniint8 &&
-                       avxvnniint16 && amxcomplex && sha512 && sm3 && sm4;
-
-            if (dmr) {
-                return "DMR (512 width)";
-            }
-        }
-        if (spr) {
-            if (__os_enabled_amx_support()) {
-                return "SPR (AMX on)";
-            } else {
-                return "SPR (AMX off)";
-            }
-        } else if (icl) {
-            return "ICL";
-        } else if (skx) {
-            return "SKX";
-        } else if (knl) {
-            return "KNL";
-        }
-        // If it's unknown AVX512 target, fall through and use AVX2
-        // or whatever is available in the machine.
-    }
-
-    if (osxsave && avx && __os_has_avx_support()) {
-        if (avx_vnni) {
-            return "AVX2VNNI (codename Alder Lake)";
-        }
-        // AVX1 for sure....
-        // Ivy Bridge?
-        if (avx_f16c && avx_rdrand) {
-            // So far, so good.  AVX2?
-            if (avx2) {
-                return "AVX2 (codename Haswell)";
-            } else {
-                // Ivy Bridge specific target was deprecated in ISPC, but
-                // no harm detecting it in standalone tool.
-                return "AVX1.1 (codename Ivy Bridge)";
-            }
+    enum ISA isa_id = get_x86_isa();
+    const char *isa = isa_strings[isa_id];
```

Comment:
If `get_x86_isa()` returns INVALID (-1), this will cause an array index out of bounds error.

Reply:
Yes, indeed. Fixed